### PR TITLE
Add handrail quest: ask if steps have a handrail or not

### LINKED ISSUE #1390: quest for steps on handrails
This quest type is useful for accessibility purposes and used into algorythm to estimate accessibility of an itinerary for blind/elderly. 
I know that it was not seen as a priority in issue #503 , but I guess it was waiting for someone to create a dedicated issue on handrails especially (issue 503 was about different quests on staircases, including handrails).
Please tell me what do you think about it.. 
cc @smortex

>> QUEST PROPOSAL (UPDATE)

### General
Affected tag(s) to be modified/added: [handrail](https://wiki.openstreetmap.org/wiki/Key:handrail)
Question asked: **Is there (an) handrail(s) along these steps?**
Answer help : Select the location of the handrail (left, right, middle).

Use case : 
1 I am an elderly, connected, I love to travel, I am still active but have moving issue, I want to know if I can go to this supermarket where there is a few steps, with no concerns..
2 I am a travel agency, I want to show accessible path to my clients. 
3 I am a city planner, I want to make my city more welcoming to people.
### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort)

### Ideas for implementation

**Proposed GUI:**
Use the same method as for sidewalks ("is there a pavement along this street), user choose the side of steps by clicking on an image/minimap and answer yes or no.

### PR DESCRIPTION
**Pull request by xuiqzy**
Please review my decision and explanation for only including `node`s and excluding only the `handrail` tag.
I tried to add a bit of documentation in the code for others to understand the rationale of how this quest is implemented (what is included/excluded etc), I hope it is done okay.

I wasn't sure about the ordering of the quest in relation to the others, maybe you have some ideas for that.

fixes #1390

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * The details would only slow down the answering in a survey by requiring more thought and a more
+ * complex GUI (interpreting the direction of the OSM way for example).
+ *
+ * Github Issue for this quest:
+ * https://github.com/westnordost/StreetComplete/issues/1390
+ */
+class AddHandrail(overpassServer: OverpassMapDataDao) : SimpleOverpassQuestType<Boolean>(overpassServer) {
+    // Do not include nodes and relations, even though these exist with the right tags, because
+    // according to the wiki page for `highway=steps` [2] it can only be applied to ways. It also
+    // does not make much sense for other types of elements and it is more likely to be a
```
Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt]: Here is something missing...

File: app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    // Do not exclude elements tagged with `handrail:left`, `handrail:center` or `handrail:right`
+    // because according to the wiki page for `handrail=*` [1], the `handrail` tag should
+    // always have just `yes` or
+    // `no` as a value to indicate the existence of a handrail somewhere. If there is a mismatch of
+    // the tags afterwards, that can be recognized using one of the various
```
Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt]: In other situations we skipped the quest in such a situation. So I would exclude `handrail:*` here as well. 
Surveyor time is valuable...

Comment (xuiqzy) [reply app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt]: So you would assume that if only `handrail:*` is tagged, the survey was valid and should not be repeated by Streetcomplete, do I understand that correctly?

File: app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt (hunk 1 above)
Comment (xuiqzy) [reply app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt]: oops, I wanted to say it is likely to be a tagging mistake. Does that make sense?

File: app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt (hunk 2 above)
Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt]: Yes

File: app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    // Quality Assurance tools [3].
+    // [1] https://wiki.openstreetmap.org/wiki/Key:handrail
+    // [2] https://wiki.openstreetmap.org/wiki/Tag:highway%3Dsteps
+    // [3] https://wiki.openstreetmap.org/wiki/Quality_assurance
+    override val tagFilters = "nodes with highway = steps and !handrail"
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt]: I think you meant to write `ways with ...`

Comment (xuiqzy) [reply app/src/main/java/de/westnordost/streetcomplete/quests/handrail/AddHandrail.kt]: yes, thanks

**Comment by xuiqzy** [comment]
Thanks for the comments @westnordost !
I cut down on the comments now. But isn't it useful to have a shortcut to the corresponding issue to the quest instead of needing to search it in the git commit history? I just thought it would make sense, since there is such a direct 1:1 mapping from implemented quest in the file to github issue here?
But I don't feel strongly about it, so if that doesn't make sense for you, I'll just remove it :)

**Comment by westnordost** [comment]
Well, depends on your tools. But any good IDE will have an annotate / history function built-in into the editor and so has Android Studio. In the editor, right-click on the bar that shows the line numbers on the left, press _annotate_. Or, to get the history, rightclick in the editor and select Git -> Show History. That is all there is.

**Comment by xuiqzy** [comment]
Okay, that is pretty easy indeed! I changed it.

**Comment by westnordost** [comment]
Looks good, thank you!

**Comment by xuiqzy** [comment]
Thanks for merging! :)
Just in case you didn't see the review request above: Was the ordering of the quests, as well as only asking it for ways okay?

**Comment by westnordost** [comment]
Yes

**Comment by versat** [comment]
It seems that it also asks for handrails if it is an escalator. I am not sure if it makes sense in this case. Just wanted to let you know. If this is by intention then everything is fine i guess.

**Comment by matkoniecz** [comment]
Yes, escalators are not excluded, thanks for reporting! (#1604 was very similar). I wonder whatever indoor stairs are also worth excluding.

**Comment by xuiqzy** [comment]
I also don't know any escalators without a handrail, but there are indoor steps with and without handrail. Why would you exclude indoor steps @matkoniecz ?

**Comment by MegaArthur** [comment]
I have definitely seen indoor steps without a handrail, @matkoniecz

**Comment by matkoniecz** [comment]
@versat I opened https://github.com/westnordost/StreetComplete/pull/1646 stopping the question for conveying steps, still asked for indoor steps.

**Pull request merged**